addons/im_livechat/static/src/public_models/public_livechat_global.js

```diff
@@ -45,8 +47,18 @@ registerModel({
         },
         async _willStart() {
             const strCookie = getCookie('im_livechat_session');
-            const isSessionCookieAvailable = Boolean(strCookie);
-            const cookie = JSON.parse(strCookie || '{}');
+            let isSessionCookieAvailable = Boolean(strCookie);
+            let cookie = JSON.parse(strCookie || '{}');
+            if (isSessionCookieAvailable && cookie.visitor_uid !== session.user_id) {
+                // livechat session is linked to another user. Either
+                // because the user was connected then disconnected or
+                // because the user was not connected and is now
+                // connected. In both cases, we should drop this
+                // session.
+                this.leaveSession();
+                isSessionCookieAvailable = false;
+                cookie = {};
+            }
```

Comment:
Comment has barely any value. It's pretty clear to understand `cookie.visitor_uid !== session.user_id` means it's not the same (authentication) for current user of livechat session in cookie. And the "drop" part is pretty obvious with few LOCs in this branching.

I would just remove the comment, or if you really want a comment it should be a single line at most.

Reply:
Removed :+1: